aws-iot-scheduledaudit/aws-iot-scheduledaudit.json

```diff
@@ -0,0 +1,134 @@
+{
+  "typeName": "AWS::IoT::ScheduledAudit",
+  "description": "Scheduled audits can be used to specify the checks you want to perform during an audit and how often the audit should be run.",
+  "sourceUrl": "https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-iot.git",
+  "definitions": {
+    "Tag": {
+      "description": "A key-value pair to associate with a resource.",
+      "type": "object",
+      "properties": {
+        "Key": {
+          "type": "string",
+          "description": "The tag's key.",
+          "minLength": 1,
+          "maxLength": 128
+        },
+        "Value": {
+          "type": "string",
+          "description": "The tag's value.",
+          "minLength": 1,
+          "maxLength": 256
+        }
+      },
+      "required": [
+        "Value",
+        "Key"
+      ]
+    }
+  },
+  "properties": {
+    "ScheduledAuditName": {
+      "description": "The name you want to give to the scheduled audit.",
+      "type": "string",
+      "pattern": "[a-zA-Z0-9:_-]+",
+      "minLength": 1,
+      "maxLength": 128
+    },
+    "Frequency": {
+      "description": "How often the scheduled audit takes place. Can be one of DAILY, WEEKLY, BIWEEKLY, or MONTHLY.",
+      "type": "string",
+      "enum": [
+        "DAILY",
+        "WEEKLY",
+        "BIWEEKLY",
+        "MONTHLY"
+      ]
+    },
+    "DayOfMonth": {
+      "description": "The day of the month on which the scheduled audit takes place. Can be 1 through 31 or LAST. This field is required if the frequency parameter is set to MONTHLY.",
+      "type": "string",
+      "pattern": "^([1-9]|[12][0-9]|3[01])$|^LAST$"
+    },
+    "DayOfWeek": {
+      "description": "The day of the week on which the scheduled audit takes place. Can be one of SUN, MON, TUE,WED, THU, FRI, or SAT. This field is required if the frequency parameter is set to WEEKLY or BIWEEKLY.",
+      "type": "string",
+      "enum": [
+        "SUN",
+        "MON",
+        "TUE",
+        "WED",
+        "THU",
+        "FRI",
+        "SAT"
+      ]
+    },
+    "TargetCheckNames": {
+      "description": "Which checks are performed during the scheduled audit. Checks must be enabled for your account.",
+      "type": "array",
+      "uniqueItems": true,
+      "items": {
+        "type": "string"
+      }
+    },
+    "ScheduledAuditArn": {
```

Comment:
Let's add length constraints (min=20 and max=2048)

Reply:
Fixed.